searx/engines/google_videos.py

```diff
@@ -17,83 +61,152 @@
+        'lr': lang_info['lr'],
+        'ie': "utf8",
+        'oe': "utf8",
+    })
+
+    if params['time_range'] in time_range_dict:
+        query_url += '&' + urlencode({'tbs': 'qdr:' + time_range_dict[params['time_range']]})
+    if params['safesearch']:
+        query_url += '&' + urlencode({'safe': filter_mapping[params['safesearch']]})
+
+    logger.debug("query_url --> %s", query_url)
+    params['url'] = query_url
+
+    logger.debug("HTTP header Accept-Language --> %s", lang_info['Accept-Language'])
+    params['headers']['Accept-Language'] = lang_info['Accept-Language']
+    params['headers']['Accept'] = (
+        'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8'
+        )
     return params
 
 
-# get response from search-request
 def response(resp):
+    """Get response from google's search request"""
     results = []
 
+    # detect google sorry
+    resp_url = urlparse(resp.url)
+    if resp_url.netloc == 'sorry.google.com' or resp_url.path == '/sorry/IndexRedirect':
+        raise SearxEngineCaptchaException()
+
+    if resp_url.path.startswith('/sorry'):
+        raise SearxEngineCaptchaException()
+
+    # convert the text to dom
     dom = html.fromstring(resp.text)
+    vidthumb_imgdata = scrap_out_thumbs(dom)
 
     # parse results
-    for result in eval_xpath_list(dom, '//div[@class="g"]'):
-
-        title = extract_text(eval_xpath(result, './/h3'))
-        url = eval_xpath_getindex(result, './/div[@class="r"]/a/@href', 0)
-        content = extract_text(eval_xpath(result, './/span[@class="st"]'))
-
-        # get thumbnails
-        script = str(dom.xpath('//script[contains(., "_setImagesSrc")]')[0].text)
-        ids = result.xpath('.//div[@class="s"]//img/@id')
-        if len(ids) > 0:
-            thumbnails_data = \
-                re.findall('s=\'(.*?)(?:\\\\[a-z,1-9,\\\\]+\'|\')\;var ii=\[(?:|[\'vidthumb\d+\',]+)\'' + ids[0],
-                           script)
-            tmp = []
-            if len(thumbnails_data) != 0:
-                tmp = re.findall('(data:image/jpeg;base64,[a-z,A-Z,0-9,/,\+]+)', thumbnails_data[0])
-            thumbnail = ''
-            if len(tmp) != 0:
-                thumbnail = tmp[-1]
-
-        # append result
-        results.append({'url': url,
-                        'title': title,
-                        'content': content,
-                        'thumbnail': thumbnail,
-                        'template': 'videos.html'})
+    for result in eval_xpath_list(dom, results_xpath):
+
+        # google *sections*
+        if extract_text(eval_xpath(result, g_section_with_header)):
+            logger.debug("ingoring <g-section-with-header>")
+            continue
+
+        title = extract_text(eval_xpath(result, title_xpath)[0])
+        url = eval_xpath(result, href_xpath)[0]
+        c_node = eval_xpath(result, content_xpath)[0]
+
+        # <img id="vidthumb1" ...>
+        img_id = eval_xpath(c_node, './div[1]//a/g-img/img/@id')
```

Comment:
lines from 181 to 184 can be:
```python
img_id = eval_xpath_getindex(result, './div[1]//a/g-img/img/@id', 0, default=None)
if img_id is None:
  continue
```

If the index 0 is not found ( = the result list is empty), `img_id` will be `None`.

Note: I've tried to find the right function signatures for the `eval_xpath_*` functions, I'm not sure if it's help : the purpose is use the same function call(s) for the same pattern.

See:
https://github.com/searx/searx/blob/f4a17acb7a4fa82c5cb629f4eaad11ef528f89e4/searx/utils.py#L583-L609